Instruments/source/CacheTupleProducer.cxx

```diff
@@ -64,31 +64,39 @@ class CacheTupleProducer {
 
         auto originalFile = root_ext::OpenRootFile(args.input_file());
         size_t n_tot_events = 0;
-        std::map<std::string,std::shared_ptr<ntuple::EventTuple>> map_event;
+        std::map<std::string,std::pair<std::shared_ptr<ntuple::EventTuple>,Long64_t>> map_event;
         for(unsigned c = 0; c < channels.size(); ++c){
             auto originalTuple = ntuple::CreateEventTuple(channels.at(c),originalFile.get(),true,ntuple::TreeState::Full);
-            map_event[channels.at(c)] = originalTuple;
-            n_tot_events += originalTuple->GetEntries();
+            const Long64_t n_events = std::min(args.max_events_per_tree(),originalTuple->GetEntries());
+            map_event[channels.at(c)] = std::make_pair(originalTuple,n_events);
+            n_tot_events += n_events;
         }
+
+        size_t n_total = n_tot_events, n_processed_events = 0;
```

Comment:
why do you create two variables with the same meaning? remove `n_total`, keep `n_tot_events`